mylyn.context/org.eclipse.mylyn.context.tasks.ui/src/org/eclipse/mylyn/internal/context/tasks/ui/editors/ContextEditorFormPage.java

```diff
@@ -231,20 +231,27 @@ public void dispose() {
 		}
 	}
 
+	private int setNewMaxWith(int oldValue, Control imageControl, Control linkControl) {
+		Point imageSize = imageControl.computeSize(SWT.DEFAULT, SWT.DEFAULT, true);
+		Point hyperlinkSize = linkControl.computeSize(SWT.DEFAULT, SWT.DEFAULT, true);
+		return imageSize.x + hyperlinkSize.x > oldValue ? imageSize.x + hyperlinkSize.x : oldValue;
+	}
+
 	private void createActionsSection(Composite composite) {
 		Section section = toolkit.createSection(composite,
 				ExpandableComposite.TITLE_BAR | ExpandableComposite.EXPANDED);
 		section.setText(Messages.ContextEditorFormPage_Actions);
 
+		int with = 0;
```

Comment:
this is an example of `temporal coupling`
in fact what we need is to ensure that after adding all the children to `sectionClient` its `widthHint` is big enough, right? So, all this code wants to be in the very end of the block, where you do
```java
sectionClientGridData.widthHint = with
```

Reply:
There is more than one link and depending on translation, it's not clear which one is wider...

Adding this to the end of the method also seemed to do the trick:
```
		sectionGridData.widthHint = section.computeSize(SWT.DEFAULT, SWT.DEFAULT).x;
```
I.e., like this on Windows (10):

![image](https://github.com/eclipse-mylyn/org.eclipse.mylyn/assets/208716/ed55b465-99f0-4c40-b1e4-e7d198836a54)

Reply:
OK, so I add some more local variables

Reply:
It really seems sufficient and much simpler to do only this:

![image](https://github.com/eclipse-mylyn/org.eclipse.mylyn/assets/208716/b49a36f5-07fa-4fc5-97dd-435928ce71b8)

I.e., let the section as a whole compute its preferred size...

Reply:
Job with this solution is https://ci.eclipse.org/mylyn/view/GitHub%20Orga/job/GitHub-mylyn-Orga/job/org.eclipse.mylyn/view/change-requests/job/PR-183/3/

When finished we should use "Squash Merge" to bring this into main.
@merks: Thanks this indeed much less code!